src/plugins/invoke/azureInvoke.ts

```diff
@@ -1,41 +1,79 @@
+import { isAbsolute, join } from "path";
 import Serverless from "serverless";
-import { join, isAbsolute } from "path";
 import AzureProvider from "../../provider/azureProvider";
+import { InvokeService } from "../../services/invokeService";
+import { AzureLoginPlugin } from "../login/loginPlugin";
+import fs from "fs";
 
 export class AzureInvoke {
   public hooks: { [eventName: string]: Promise<any> };
+  private commands: any;
   private provider: AzureProvider;
-
+  private login: AzureLoginPlugin
+  private invokeService: InvokeService;
   public constructor(private serverless: Serverless, private options: Serverless.Options) {
```

Comment:
Consider creating a new interface for `InvokeOptions` so we can make this more strongly typed and avoid using the square braket syntax.